backend/src/data/repositories/user/user.repository.ts

```diff
@@ -11,8 +15,17 @@ class User {
     this.#UserModel = UserModel;
   }
 
-  async getAll(): Promise<UserM[]> {
-    return this.#UserModel.query();
+  async getPaginated({
+    page,
+    count,
+  }: EntityPaginationRequestQueryDto): Promise<EntityPagination<UserM>> {
+    const ZERO_INDEXED_PAGE = page - 1;
```

Comment:
This should be done in service - it can know what need to be modified to bind request with data retrieval.